Pat Mifsud/week01/homework/day4_train/script.js

```diff
@@ -0,0 +1,271 @@
+
+
+// MTA Lab
+// 🚨 as an added challange i tried to also calculate price for the trip, with a different price per line, and display the platform for each station. I regret doing that.
+
+
+// ------------------------------------
+// Initalising things
+
+// Array to log the trip for latter reporting
+let log = [];
+
+// Counter
+let stopsMade = 0;
+
+
+// ------------------------------------
+// Stations, stops and lines
+const subway = {
+//    name           platform#
+  lineN : {
+    stop : [
+      [ "Times Square" , 4 ],
+      [ "34th Street" , 1 ],
+      [ "28th Street" , 1 ],
+      [ "23rd Street" , 1 ],
+      [ "Union Square" , 9 ],
+      [ "8th Street" , 1 ]
+    ],
+    price : 1.25,
+    direction: "north"
+  },
+  lineL : {
+    stop : [
+      [ "8th Street" , 2 ],
+      [ "6th Street" , 1 ],
+      [ "Union Square" , 8 ],
+      [ "3rd Street" , 1 ],
+      [ "1st Street" , 1 ]
+    ],
+    price : 0.50,
+    direction: "east"
+  },
+  line6 : {
+    stop : [
+      [ "Grand Central" , 26 ],
+      [ "33rd Street" , 1 ],
+      [ "28th Street" , 2 ],
+      [ "23rd Street" , 2 ],
+      [ "Union Square" , 7 ],
+      [ "Astor Place" , 1 ]
+    ],
+    price : 3.10,
+    direction: "west"
+  },
+};
+
+
+// --------------------------
+// 🏭  Factories
+
+
+const formatStops = function(lineStops){
+  let formattedArray = [];
+  for (let i = 0; i < lineStops.length; i++){
+    formattedArray.push({
+      name: lineStops[i][0],
+      platform: lineStops[i][1]
+    });
+  };
+return formattedArray;
+}
+
+
+const formatSubwayObj = function(){
+  for (Object in subway){
```

Comment:
You should avoid using `Object` keyword as a variable's name since it's a special keyword in JavaScript.